sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Primitives;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample10_MockingReadingEventsTests
+    {
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReadPartition()
+        {
+            #region Snippet:EventHubs_Sample05_MockingEventHubConsumer
+
+            var mockEventHubConsumerClient = new Mock<EventHubConsumerClient>();
+            var mockPartitionId = "sample partition id";
+
+            mockEventHubConsumerClient
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[] { mockPartitionId });
+            mockEventHubConsumerClient
+                .Setup(c => c.ReadEventsFromPartitionAsync(It.IsAny<string>(), It.IsAny<EventPosition>(), It.IsAny<CancellationToken>()))
+                .Returns(GetMockEventsFromPartition(mockPartitionId));
```

Comment:
I see what you mean, it'd be better to keep it to a minimum, so that it's easier to understand.
I could always write an overload to the `GetMockEvents()` with last enqueued event and use that for a new sample.

Reply:
However, I dont see the `EventData` constructor which accepts string:
![image](https://user-images.githubusercontent.com/25906933/124164941-e674fc80-dabe-11eb-86b2-5bfe08c5eb8c.png)

Reply:
![image](https://user-images.githubusercontent.com/25906933/124165041-07d5e880-dabf-11eb-88aa-62a29f519ad7.png)

Reply:
Rebase on `main`, it's a recent commit.  😄

Reply:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs#L419

Reply:
updated the function to generate `EventData` inline

Reply:
Cool, it looks good!   Now we just need to hoist it into the snippet as a local function so that everything is available to the reader at a glance.